### PR TITLE
Adding the ability to deploy the resources in an MFA subscription

### PR DESCRIPTION
**Pull request by dcaro**
## Purpose
Those proposed changes allows the deployment of the DevOps OpenHack resources into a subscription that is using multi factor authentication. 

## Does this introduce a breaking change?
```
[X] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Follow the instructions in the `provision-vm/readme.md` file. 

## What to Check
Verify that the following are valid
* The Azure resources are deployed correctly 
* The APIs are deployed and running

### CODE REVIEW

File: provision-vm/azuredeploy.json [hunk 1]
```diff
@@ -225,11 +239,17 @@
                 "autoUpgradeMinorVersion": true,
                 "settings": {
                     "fileUris": [
-                        "https://raw.githubusercontent.com/Azure-Samples/openhack-devops-proctor/master/provision-vm/proctorVMSetup.sh"
+                        "https://raw.githubusercontent.com/Azure-Samples/openhack-devops-proctor/feature/selfguided/provision-vm/proctorVMSetup.sh"
```
Comment (rguthriemsft) [inline provision-vm/azuredeploy.json]: is this the right repo url?  Assume once in you will need to send another PR to set to master?

File: provision-team/provision_aks.sh [hunk 2]
```diff
@@ -97,7 +115,7 @@ kvstore set ${teamName} ACR_URI $ACR_ID
 echo "Granting Service Princpal " $SP_NAME " access to ACR " $teamName"acr" "..."
 (
     set -x
-    az role assignment create --assignee $SP_ID --role Reader --scope $ACR_ID
+    az role assignment create --assignee $SP_ID --role acrpull --scope $ACR_ID
```
Comment (rguthriemsft) [inline provision-team/provision_aks.sh]: Is this correct?  https://docs.microsoft.com/en-us/azure/container-registry/container-registry-auth-aci#create-a-service-principal The doc seems to indicate this can be set during Service Principal creation.  Not sure what is best method.

File: provision-vm/proctorVMSetup.sh [hunk 3]
```diff
@@ -53,7 +55,7 @@ touch /home/azureuser/.bashrc
 echo 'export PATH=$PATH:/opt/mssql-tools/bin' >> /home/azureuser/.bashrc
 
 echo "############### Pulling Openhack-tools from Github ###############"
-sudo git clone https://github.com/Azure-Samples/openhack-devops-proctor.git /home/azureuser/openhack-devops-proctor
+sudo git clone https://github.com/Azure-Samples/openhack-devops-proctor.git --branch feature/selfguided /home/azureuser/openhack-devops-proctor
```
Comment (rguthriemsft) [inline provision-vm/proctorVMSetup.sh]: Same comment as above, should this be master?

File: provision-team/provision_aks.sh (hunk 2 above)
Comment (dtzar) [reply provision-team/provision_aks.sh]: It can, but there are bugs with AKS + ACR auth so this is why we did it this way.
https://github.com/Azure/AKS/issues?utf8=%E2%9C%93&q=is:issue+is:open+ACR
